### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by ColinMaudry**
Extension update to match the new mapping for subcontracting.

### CODE REVIEW

**Comment by ColinMaudry** [comment]
I'm merging master in develop, fixing the conflicts.

**Comment by ColinMaudry** [comment]
@jpmckinney Ready for review :sweat_smile:

File: release-schema.json [hunk 1]
```diff
@@ -50,10 +50,38 @@
           "maximum": 1,
           "minimum": 0
         },
+        "competitiveMaximumPercentage": {
+          "title": "Maximum percentage of total value for competitive subcontracting",
```
Comment (jpmckinney) [inline release-schema.json:54]: Should it be "contract value" instead of "total value"? Same below.

File: release-schema.json [hunk 2]
```diff
@@ -50,10 +50,38 @@
...
+          "description": "Maximum percentage of the contract value the procuring entity asks the contractor to subcontract to third parties through a competitive procedure.",
```
Comment (jpmckinney) [inline release-schema.json:55]: Is it "asks" or "requires"? Also, start with "The " and add "that" before "the procuring entity", to make it easier to parse the sentence into clauses.

File: release-schema.json [hunk 3]
```diff
@@ -50,10 +50,38 @@
...
+          "type": [
+            "number",
+            "null"
+          ],
+          "maximum": 1,
+          "minimum": 0.0001
```
Comment (jpmckinney) [inline release-schema.json:61]: Instead of 0.0001, we can use `"exclusiveMinimum": true`. Same above.

File: release-schema.json [hunk 4]
```diff
@@ -50,10 +50,38 @@
...
+          "minimum": 0
+        },
         "value": {
           "title": "Subcontracted value",
           "description": "The estimated value of the part of the contract that the contractor will subcontract to third parties.",
           "$ref": "#/definitions/Value"
+        },
+        "competitive": {
+          "title": "Competitive subcontracting",
+          "description": "Whether the contractor will subcontract part of the contract to third parties through a competitive procedure.",
```
Comment (jpmckinney) [inline release-schema.json:80]: Should we replace "contractor" with "supplier" throughout, to match OCDS terms?

Comment (ColinMaudry) [reply release-schema.json:80]: That makes sense, yes.

File: release-schema.json (hunk 2 above)
Comment (ColinMaudry) [reply release-schema.json:55]: [Article 21, 4) ](https://eur-lex.europa.eu/legal-content/EN/TXT/HTML/?uri=CELEX:32009L0081&from=EN#d1e2623-76-1)

> Member States may provide that the contracting authority/entity may ask or be required to **ask** the successful tenderer to subcontract to third parties a share of the contract.

But also:

> The successful tenderer shall award subcontracts corresponding to the percentage which the contracting authority/entity requires it to subcontract in accordance with the provisions of Title III.

I guess it boils down to a requirement. I wish EU lawmakers stuck to may/could/should/must, as in W3C recommendations!

File: release-schema.json (hunk 1 above)
Comment (ColinMaudry) [reply release-schema.json:54]: I meant total contract value, but indeed contract value is clearer.

Comment (ColinMaudry) [reply release-schema.json]: Done

File: README.md [hunk 5]
```diff
@@ -8,7 +8,7 @@ If the percentage of the contract value that is subcontracted is an exact number
 
 ## Legal context
 
-In the European Union, this extension's fields correspond to [eForms BG-180 (Subcontracting)](https://github.com/eForms/eForms). See [OCDS for the European Union](http://standard.open-contracting.org/profiles/eu/master/en/) for the correspondences to Tenders Electronic Daily (TED).
+In the European Union, this extension's fields correspond to [eForms BG-709 (Second Stage)](https://github.com/eForms/eForms) and [article 21 of directive 2009/81/EC](https://eur-lex.europa.eu/legal-content/EN/TXT/HTML/?uri=CELEX:32009L0081&from=EN#d1e2623-76-1). See [OCDS for the European Union](http://standard.open-contracting.org/profiles/eu/master/en/) for the correspondences to Tenders Electronic Daily (TED).
```
[the remark this answers is not in the document]
Comment (ColinMaudry) [reply README.md:11]: Yes we should. I'll add it.

File: release-schema.json (hunk 3 above)
Comment (ColinMaudry) [reply release-schema.json]: I get an error during validation when I add `exclusiveMinimum`:

> Additional properties are not allowed ('exclusiveMinimum' was unexpected) (properties/definitions/additionalProperties/properties/properties/additionalProperties/additionalProperties)

It also results in an error if add `exclusiveMinimum` without `minimum`.

Comment (ColinMaudry) [reply release-schema.json]: https://github.com/open-contracting-extensions/ocds_subcontracting_extension/pull/3/checks?check_run_id=554900208

File: release-schema.json (hunk 2 above)
Comment (ColinMaudry) [reply release-schema.json]: Done

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (ColinMaudry) [reply README.md]: Done

File: README.md [hunk 6]
```diff
@@ -46,10 +47,22 @@ In the European Union, this extension's fields correspond to [eForms BG-180 (Sub
...
+        "competitiveMinimumPercentage": 0.1,
+        "competitiveMaximumPercentage": 0.25,
+        "description": "The painting and electricity tasks are subcontracted."
+      }
+    }
+  ]
+}
+```
 
-## Changelog
```
[the remark this answers is not in the document]
Comment (ColinMaudry) [reply README.md:53]: Accidentally yes. I fixed it.

File: release-schema.json (hunk 3 above)
Comment (ColinMaudry) [reply release-schema.json]: That's the last thing to fix in this PR.

Comment (jpmckinney) [reply release-schema.json]: Thanks - it's because this was the first time we used that property, so I had to relax our checks.

**Pull request merged**